src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/DefaultLSPRequestInvoker.cs

```diff
@@ -117,5 +120,15 @@ public DefaultLSPRequestInvoker(ILanguageClientBroker languageClientBroker)
             var result = resultToken != null ? resultToken.ToObject<TOut>(_serializer) : default;
             return result;
         }
+
+        internal override void AddClientNotifyAsyncHandler(AsyncEventHandler<LanguageClientNotifyEventArgs> clientNotifyAsyncHandler)
```

Comment:
Waittt so `clientNotifyAsyncHandler` doesn't take any sort of cancellation token?

Reply:
Nope. Now that you mention it, `clientNotifyAsyncHandler` would be better described as a listener than a handler.

It's responsible for simply listening to any `LanguageClientNotifyEventArgs` notification event sent by VS LSP / Roslyn to our LanguageServerClient. There's no cancellation token as it isn't for any specific request.